shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -2769,6 +2765,33 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
             exists(Ap ap |
               localFlowStepPlus(node1, state1, node2, preservesValue, t, callContext, label) and
               localFlowExit(node2, state1, ap) and
+              state1 = state2 and
+              node1 != node2
```

Comment:
It makes a difference, since otherwise making the mere `localFlowEntry/Exit` restriction on an existing TC will not 
necessarily produce the same as recomputing TC based on the existing TC. This is because there is a `node1 != node2` restriction in the initial step of the TC computation logic.